python/cuml/svm/svc.pyx

```diff
@@ -460,8 +460,14 @@ class SVC(SVMBase,
                                                cv=5,
                                                method='sigmoid')
 
+        # Apply class weights to sample weights, necessary so it doesn't crash when sample_weight is None
+        # could also be done in .fit()
+        sample_weight = apply_class_weight(self.handle, sample_weight, self.class_weight, y, self.verbose,
+                                           self.output_type, self.dtype)
+
         with cuml.internals.exit_internal_api():
-            self.prob_svc.fit(X, y)
+            # .get() is necessary because sklearn requires explicit conversion to np array
+            self.prob_svc.fit(X, y, sample_weight=sample_weight.get())
```

Comment:
The tests revealed a potential issue with the fact that sample_weight can be none here, so we need to deal with that.

Reply:
You/the test were right! I added a condition to fix this. I also noticed a small existing issue during my tests, where trying to fit a probabilistic SVC with class weights and less than 3 classes (see below), would raise a weird error, so I added additional and informative error handling.

```
X_train = np.random.rand(1000, 100).astype(np.float32)
# 2 Classes
y_train = np.random.randint(0, 2, 1000)
# Probabilistic SVC with class weights
svc = SVC(C=1, probability=True, kernel='rbf', class_weight='balanced')
# If we try to fit this, we get an error.
svc.fit(X_train, y_train) # ValueError: Buffer dtype mismatch, expected 'const float' but got 'double'
```